src/NextGen/batchManager.ts

```diff
@@ -0,0 +1,52 @@
+import {
+    Batch,
+    BatchEnvironmentEnum,
+    BaseEvent,
+} from '@mparticle/event-models';
+import { BaseMPMessage } from './core/baseMPMessage';
+import { isEmpty } from '../utils';
+
+// TODO: Sync this with existing API Client(s)
+export interface IMpApiClientImplementation {
+    uploadBatch(batch: Batch): void;
+}
+
+export default class BatchManager {
```

Comment:
@markvdouw I thought that was the job of the mediator, to take data from one component to another.

Reply:
The mediator should be juggling with the EXTERNAL public components (Identity, Consent, Media, EventLogging, etc), that customer will basically relate to our public API. 
Though the BatchHandler can be considered one of the main components, it should, in my opinion, delegate data handling. One of the patterns widely used for data sources handling is the Repository Pattern (its basically a handler/manager for multiple data sources such as DAOs, Cache, ApiClients, etc.)

I would advice injecting a data repository in every component that needs to interact with data. 
This is THE case of data manipulation (writing into the database, reading from it/creating a batch and pass it to the apiClient so it can be updated, etc)